src/main/scala/cromwell/binding/AstTools.scala

```diff
@@ -29,19 +29,35 @@ object AstTools {
             case WdlFloatType.toWdlString => WdlFloatType
             case WdlBooleanType.toWdlString => WdlBooleanType
             case WdlObjectType.toWdlString => WdlObjectType
+            case _ => throw new SyntaxError(wdlSyntaxErrorFormatter.arrayHasZeroTypeParameters(t))
           }
         case a: Ast =>
           val subtypes = a.getAttribute("subtype").asInstanceOf[AstList].asScala.toSeq
           val typeTerminal = a.getAttribute("name").asInstanceOf[Terminal]
           a.getAttribute("name").sourceString() match {
             case "Array" =>
-              val member = subtypes.head.wdlType
+              if (subtypes.size != 1) throw new SyntaxError(wdlSyntaxErrorFormatter.arrayHasOneTypeParameter(typeTerminal))
```

Comment:
The method naming here is a bit confusing

`arrayHasZeroTypeParameters` implies it's a problem the array is specified with no type parameters.

By symmetry one would think `arrayHasOneTypeParameter` complains the array is specified with one type parameter, but actually it means it's a problem the array was specified with more than one.